### PR TITLE
enhancements: Remove "[optional]" from headers

### PR DESCRIPTION
**Pull request by wking**
These mark optional sections in the template, but are no longer
required in actual enhancements where the decision has been made to
use the given section.  Generated with:

```console
$ sed -i 's/ \[optional]$//' $(git grep -l '\[optional]')
$ git checkout HEAD enhancements/template.md
```

### CODE REVIEW

**Comment by russellb** [comment]
@wking This looks like an easy quick merge, but it'll need to be redone.  If you want to update this, ping me and I'll approve it so we can close it out.

**Comment by wking** [comment]
Regenerated from scratch on master with 8d3c2cd -> eabf021.

File: enhancements/install/user-provided-vpcs.md [hunk 1]
```diff
@@ -0,0 +1,184 @@
+---
+title: user-provided-vpcs-on-aws
```
Comment (russellb) [inline enhancements/install/user-provided-vpcs.md:2]: I don't think you meant to include this here?

File: enhancements/update/cluster-version-operator-x509-upstream-trust.md [hunk 2]
```diff
@@ -0,0 +1,230 @@
+---
+title: cluster-version-operator-x.509-upstream-trust
```
Comment (russellb) [inline enhancements/update/cluster-version-operator-x509-upstream-trust.md:2]: or this one?

File: enhancements/install/user-provided-vpcs.md (hunk 1 above)
Comment (wking) [reply enhancements/install/user-provided-vpcs.md:2]: nope.  Removed with eabf021 -> 3fe8a38.

File: enhancements/update/cluster-version-operator-x509-upstream-trust.md (hunk 2 above)
Comment (wking) [reply enhancements/update/cluster-version-operator-x509-upstream-trust.md:2]: nope.  Removed with eabf021 -> 3fe8a38.

**Pull request merged**